packages/mui-material/src/ListItemButton/ListItemButton.js

```diff
@@ -168,7 +168,7 @@ const ListItemButton = React.forwardRef(function ListItemButton(inProps, ref) {
     <ListContext.Provider value={childContext}>
       <ListItemButtonRoot
         ref={handleRef}
-        component={component}
+        component={other.href && component === 'div' ? 'a' : component}
```

Comment:
```suggestion
        component={(other.href || other.to) && component === 'div' ? 'a' : component}
```

How about we consider the `to` prop too. We are doing it in the [ButtonBase](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/ButtonBase/ButtonBase.js#L293) component

Reply:
We will also need to add tests for this prop too.

Reply:
Sounds good